### ISSUE TITLE
intentionally empty generator marked as unreachable

### ISSUE THREAD

**Issue by asottile** [python/mypy#15345]
**Bug Report**

this seems to be a common pattern:

```python
def f() -> Generator[None, None, None]:
    return
    yield
```

it can instead be rewritten as:

```python
def f() -> Generator[None, None, None]:
    yield from ()
```

**To Reproduce**

see above

**Expected Behavior**

(no errors)

**Actual Behavior**

```console
$ mypy --warn-unreachable t.py
t.py:5: error: Statement is unreachable  [unreachable]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 1.3.0
- Mypy command-line flags: `--warn-unreachable`
- Mypy configuration options from `mypy.ini` (and other config files): N/A
- Python version used: 3.10.6

**Comment by Hayashi-Yudai** [comment]
Uum... I couldn't reproduce this bug.

- mypy: 1.3.0
- with `--warn-unreachable` flag
- w/o `mypy.ini`
- python 3.10.5

```bash
❯ mypy --warn-unreachable t.py
Success: no issues found in 1 source file

❯ cat t.py
from typing import Generator


def f() -> Generator[None, None, None]:
    yield from ()

```

**Comment by asottile** [comment]
yes you've used the workaround.

**Comment by ikonst** [comment]
What could be a good heuristic here? That an empty yield statement (without expression) is explicitly allowed to trail a return statement?

**Comment by asottile** [comment]
I'm not sure -- I think the only case is specifically the one in the issue -- I can't think of others since if it were conditional then there'd probably be a yield in the other arm.  maybe this is specifically a cutout for `return; yield` as the total body?

**Comment by ikonst** [comment]
How about every function having a budget for one bare yield that doesn't get flagged?

**Comment by ikonst** [comment]
Planning to address it on top of #15386.
